pkg/backup/pruner.go

```diff
@@ -0,0 +1,65 @@
+package backup
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Prune prunes the directory keeping the n most recent backup files.
+func Prune(dir string, keep int) error {
+	if keep < 1 {
+		return errors.New("keep less than one")
+	}
+
+	files, err := readBackupFiles(dir)
+	if err != nil {
+		return fmt.Errorf("reading backup files: %s", err)
+	}
+
+	if len(files) <= keep {
+		return nil
+	}
+
+	toBeRemoved := files[:(len(files) - keep)]
+	for _, file := range toBeRemoved {
+		if err := os.Remove(path.Join(dir, file.Name())); err != nil {
+			return errors.Errorf("os remove: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func readBackupFiles(dir string) ([]fs.FileInfo, error) {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		return []fs.FileInfo{}, fmt.Errorf("read dir: %s", err)
+	}
+
+	backupFiles := []fs.FileInfo{}
+	for _, f := range files {
```

Comment:
This loop filters to consider only backup files, that is files that start with `tbl_backup` and end with `.db` or `.db.gz`.

This is to accidentally not remove any other kind of file.

Reply:
This is a good point. Specially if we add a pruning CLI command that can be run at the same time as the background routine.